Chapter 4 framework (#2200)
cntt-n/CNTT: doc/ref_cert/RC2/chapters/chapter04.md

# 4. CNF Test Cases and Requirements Traceability

## Table of Contents
* [4.1 Introduction](#4.1)
* [4.2 Selection Criteria](#4.2)
* [4.3 Traceability Matrix](#4.3)

>   * [4.3.1Test Case Traceability to RA2 Requirements](#4.3.1)

Reviewer: missing blank char

>   * [4.3.1Test Case Traceability to RA2 Requirements](#4.3.1)

Author: ```suggestion
  * [4.3.1 Test Case Traceability to RA2 Requirements](#4.3.1)
```

>   * [4.3.1 Test Case Traceability to CNF Interoperability Requirements](#4.3.1)

Reviewer 2: "CNF Interoperability" may be confusing since RA2 doesn't specify any CNF interoperability (between CNFs)? The aspects of CNFs to test are for portability i.e. CNF can run on different infrastructures

Author: from #1496 "Clear position agreed between CNTT and CNCF: RA2 shoudl only deal with requirements that workloads have in order to interoperate with an RA2-compliant platform. Cloud native best practices and the requirements relating to those will be handled in either the CNCF TUG or CNF Working Group."

Author: I would say it is interoperability between the CNF and the platform, not between CNFs. Does that make sense? Or is there a different way you would phrase it?

Reviewer 2: It makes sense if you already know the details of position agreed between CNTT vs CNCF. There has been lots of confusion about that so we should try to clarify where possible. Either les make it explicit or perhaps "CNF portability" may be more appropriate?

>   * [4.3.1 Test Case Traceability to CNF Interoperability Requirements](#4.3.1)

Author: ```suggestion
  * [4.3.1Test Case Traceability to RA2 Requirements](#4.3.1)
```

  
<a name="4.1"></a>
## 4.1 Introduction

> The scope of this chapter is to identify and list down test cases based on requirements defined in [Reference Architecture-2 (RA-2)](../../../ref_arch/kubernetes/README.md). This will serve as traceability between test cases and requirements for CNF interoperability.

Reviewer: It's more for Kubernetes interoperability or CNF onboarding

> The scope of this chapter is to identify and list down test cases based on requirements defined in [Reference Architecture-2 (RA-2)](../../../ref_arch/kubernetes/README.md). This will serve as traceability between test cases and requirements for CNF interoperability.

Author: ```suggestion
The scope of this chapter is to identify and list down test cases based on requirements defined in [Reference Architecture-2 (RA-2)](../../../ref_arch/kubernetes/README.md). This will serve as traceability between test cases and requirements for Kubernetes platform interoperability.
```

> The scope of this chapter is to identify and list down test cases based on requirements defined in [Reference Architecture-2 (RA-2)](../../../ref_arch/kubernetes/README.md). This will serve as traceability between test cases and requirements for Kubernetes platform interoperability.

Reviewer 3: nit
```suggestion
The scope of this chapter is to identify and list test cases based on requirements defined in [Reference Architecture-2 (RA-2)](../../../ref_arch/kubernetes/README.md). This will serve as traceability between test cases and requirements for Kubernetes platform interoperability.
```

> The scope of this chapter is to identify and list down test cases based on requirements defined in [Reference Architecture-1 (RA-2)](../../../ref_arch/kubernetes/README.md). This will serve as traceability between test cases and requirements for CNF interoperability.

Reviewer 4: ```suggestion
The scope of this chapter is to identify and list down test cases based on requirements defined in [Reference Architecture-2 (RA-2)](../../../ref_arch/kubernetes/README.md). This will serve as traceability between test cases and requirements for CNF interoperability.
```

> The scope of this chapter is to identify and list down test cases based on requirements defined in [Reference Architecture-2 (RA-2)](../../../ref_arch/kubernetes/README.md). This will serve as traceability between test cases and requirements for CNF interoperability.

Reviewer 2: requirements relevant to CNF use of infrastructure

Author: from #1496 "Clear position agreed between CNTT and CNCF: RA2 shoudl only deal with requirements that workloads have in order to interoperate with an RA2-compliant platform. Cloud native best practices and the requirements relating to those will be handled in either the CNCF TUG or CNF Working Group."

Reviewer 2: See my comment above https://github.com/cntt-n/CNTT/pull/2200#discussion_r563931220

Reviewer 3: In the meeting on January 28, 2021, it was decided that the title "[4.3.1 Test Case Traceability to CNF Interoperability Requirements](https://github.com/cntt-n/CNTT/blob/Chapter-4-framework/doc/ref_cert/RC2/chapters/chapter04.md#43-traceability-matrix)" to "4.3.1 Test Case Traceability to RA2 Requirements"

Note that each requirement may have one or more test cases associated with it.

<a name="4.2"></a>
## 4.2 Selection Criteria

> Test cases and their dependencies must be open source. The test cases (or test suite with the test case) as well as the environment needed to run the test should be reproducible by any party following publicly available documentation.

Author: ```suggestion
Test cases, tools and their dependencies must be open source. The test cases (or test suite with the test case) as well as the environment needed to run the test should be reproducible by any party following publicly available documentation.
```

> Test cases and their dependencies must be open source. The test cases (or test suite with the test case) as well as the environment needed to run the test should be reproducible by any party following publicly available documentation.

Reviewer 3: "Test cases and their dependencies must be open source" -- is it test cases to be open source or Test Tools?

> Examples of initiatives (having testing tools, test suites, etc) with test cases which could be used include K8s Conformance, K8s e2e, Sonobuoy, Anuket FuncTest, CNF Conformance.

Reviewer: Functest (note: CNF Conformance is integrated by Functest - the overall CI integration)

> Examples initiatives (having testing tools, test suites, etc) with test cases which could be used include K8s Conformance, K8s e2e, Sonobuoy, OPNFV FuncTest, CNF Conformance.

Reviewer 4: ```suggestion
Examples of initiatives (having testing tools, test suites, etc) with test cases which could be used include K8s Conformance, K8s e2e, Sonobuoy, Anuket FuncTest, CNF Conformance.
```

> Examples of initiatives (having testing tools, test suites, etc) with test cases which could be used include K8s Conformance, K8s e2e, Sonobuoy, Anuket FuncTest, CNF Conformance.

Author: ```suggestion
Examples of initiatives (having testing tools, test suites, etc) with test cases which could be used include K8s Conformance, K8s e2e, Sonobuoy, Anuket Functest, CNF Conformance.
```

<a name="4.3"></a>
## 4.3 Traceability Matrix

The following is a Requirements Traceability Matrix (RTM) mapping Test Case, and/or Test Case Coverage, to RM and RA requirements -- configuration, deployment, runtime.

<a name="4.3.1"></a>

> ### 4.3.1 Test Case Traceability to CNF Interoperability Requirements

Author: ```suggestion
### 4.3.1 Test Case Traceability to RA2 Requirements
```

This section focuses on the test case covering requirements in [RA-2 Chapter 4.9](https://github.com/cntt-n/CNTT/blob/master/doc/ref_arch/kubernetes/chapters/chapter04.md#49-kubernetes-workloads) for Kubernetes workloads.

| RM/RA Ref | High-level test definition | Test name and project | Priority |
|---|---|---|---|

> | ra2.app.001 | | | Must |

Reviewer: I don't see the point to add blank tabs.

Author: They will be filled in later

Author: the requirements are already in RA2
